### PR TITLE
add Matrix and Mastodon to communities section #767

### PR DESCRIPTION
**Pull request by sabine**
Addresses #767. Also expands text on some options.

Please check that links and text is as expected @avsm @talex5 @Swrup. (I haven't used these services yet, so I took my best guess at how to invite people to these servers.)

|before|after|
|-|-|
|Screenshot 2023-01-12 at 17-46-33 The OCaml Community|Screenshot 2023-01-17 at 09-39-59 The OCaml Community|

### CODE REVIEW

**Comment by talex5** [comment]
I'm not sure asking everyone to join types.pl is a good idea. It looks fairly exclusive (815 users, and you have to write an application and wait for it to be reviewed).

Do we have an official OCaml Mastodon account somewhere? It would probably better to link to that if so. Then they have some stuff to read, and can decide to sign up somewhere if they want. For example, my profile is https://mastodon.social/@talex5 (don't link to that though - it's just an example of the kind of thing they'd see).

It also seems a little odd to describe it as "Like Twitter, but Open Source". If we prefer open systems, perhaps we should list it first and describe Twitter as "Like Mastodon, but closed source, non-federated, and controlled by a single entity"?

The Matrix link looks fine to me.

**Comment by sabine** [comment]
Point taken. Introducing Mastodon as "Join the community on Mastodon, a decentralized social network." instead. Screenshot updated.

I'm waiting on community input on whether there's a better Mastodon server or even an official OCaml account on Mastodon to link to. If we don't reach consensus by next Wednesday, I'll merge the Matrix commit and Mastodon can be added later, when we know where to point it.

**Comment by avsm** [comment]
Right now, we do not have an official OCaml language mastodon account (for that matter, we don't have an official Twitter one either).

We could set up an OCaml account on one of the bigger Mastodon servers (ocaml@mastodon.social appears to be free, for example).  If it becomes popular, then we could migrate it to @ocaml.org in the future, as Mastodon supports account migration.

File: src/ocamlorg_frontend/pages/community.eml [hunk 1]
```diff
@@ -37,27 +37,35 @@ Layout.render
...
           <div class="font-normal text-xs mt-4">Chat with other members of the community on Discord.</div>
         </a>
         <a href="https://twitter.com/ocamllang" class="border-gray-200 border p-8 rounded-xl card-hover">
           <div class="text-base font-semibold"><img src="/img/community/twitter.svg" alt="" class="h-8 w-8 mr-2 inline-block" /> Twitter</div>
-          <div class="font-normal text-xs mt-4">Catch up with some of the latest announcement and events from the community.</div>
+          <div class="font-normal text-xs mt-4">Catch up with some of the latest announcements and events from the community.</div>
+        </a>
+        <a href="https://types.pl/" class="border-gray-200 border p-8 rounded-xl card-hover">
+          <div class="text-base font-semibold"><img src="/img/community/mastodon.svg" alt="" class="h-8 w-8 mr-2 inline-block" /> Mastodon</div>
+          <div class="font-normal text-xs mt-4">Join the community on Mastodon, a decentralized social network. The types.pl Mastodon homeserver is popular among the OCaml community.</div>
```
Comment (avsm) [inline src/ocamlorg_frontend/pages/community.eml]:
```suggestion
          <div class="font-normal text-xs mt-4">Join the community on Mastodon, a decentralized social network. The <a href="https://types.pl/">types.pl</a> or <a href="https://discuss.systems/">discuss.systems</a> Mastodon homeservers are both popular choices among the OCaml community.</div>
```

File: src/ocamlorg_frontend/pages/community.eml [hunk 2]
```diff
@@ -37,27 +37,35 @@ Layout.render
...
         </a>
         <a href="irc://irc.libera.chat/#ocaml" class="border-gray-200 border p-8 rounded-xl card-hover">
           <div class="text-base font-semibold"><img src="/img/community/hashtag.svg" alt="" class="h-8 w-8 mr-2 inline-block" /> IRC Chat</div>
-          <div class="font-normal text-xs mt-4">Chat with other members of the community on IRC.</div>
+          <div class="font-normal text-xs mt-4">Chat with other members of the community at #ocaml on the Libera Chat IRC network.</div>
+        </a>
+        <a href="https://matrix.to/#/#ocaml-language:matrix.org" class="border-gray-200 border p-8 rounded-xl card-hover">
```
Comment (avsm) [inline src/ocamlorg_frontend/pages/community.eml]: Let's link to the OCaml space instead: https://matrix.to/#/#ocaml:roscidus.com -- that one adds other rooms in addition to the OCaml language chat one.

File: src/ocamlorg_frontend/pages/community.eml (hunk 1 above)
Comment (sabine) [reply src/ocamlorg_frontend/pages/community.eml]: We're already in a HTML `<a>`-tag here that links to something, so putting two links inside that won't work (the small links will be super small and hard to click within the huge box which is already a link). So, either we can link to Mastodon in general, and then mention these servers, or choose one to primarily link to and mention the other in the text.

Comment (sabine) [reply src/ocamlorg_frontend/pages/community.eml]: That or have two Mastodon boxes, one for each server.

Comment (avsm) [reply src/ocamlorg_frontend/pages/community.eml]: not making them linkable should be fine, but I think it's important to mention two possible servers.

**Comment by sabine** [comment]
Matrix box links to https://matrix.to/#/#ocaml:roscidus.com.

Mastodon box is now not a link but contains links to the two homeservers https://discuss.systems/ and https://types.pl/.

IMO ready to merge.

**Review by cuihtlauac — APPROVED** [review]

**Comment by redianthus** [comment]
@sabine why are we linking to roscidus.com ? There's a proper matrix canal with 200+ members and it's at #ocaml-lang:matrix.org ; it has been there for a long time...

**Comment by sabine** [comment]
@zapashcanon https://github.com/ocaml/ocaml.org/pull/796#discussion_r1070417876 I know too little about Matrix to see what's the better link.

**Comment by talex5** [comment]
@zapashcanon the link is to the OCaml space, not to the individual channel. But it would indeed be good to publish that at matrix.org too - perhaps someone with an account there could add it?

**Comment by redianthus** [comment]
There is `#ocaml-space:matrix.org` if that's what you mean ? I added it in #1527.

**Pull request merged**